apps/crawler/crawler/s3/tests/unit/test_s3_crawler.py

```diff
@@ -34,7 +34,7 @@ def mock_download_func(bucket, key, file):
         download.side_effect = mock_download_func
 
     def test_s3_crawler_first_crawl(self, mocker, tmp_path: Path):
-        s3_crawler = S3Crawler("bucket", "prefix")
+        s3_crawler = S3Crawler("bucket", "prefix", False)
```

Comment:
I'd expect you to have to change something in the mocking to make sure that you either require credentials or not.

Reply:
I wasn't going to worry about this right now.  It's still testing what it was meant to test.

Reply:
`# TODO: Add a test to verify that this works in anonymous mode. Probably needs to be an integration test to be useful`